solcjs

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env node
+
+var fs = require('fs');
+var solc = require('./index.js');
+// FIXME: remove annoying exception catcher of Emscripten
+//        see https://github.com/chriseth/browser-solidity/issues/167
+process.removeAllListeners('uncaughtException');
+
+var yargs = require('yargs')
+  .usage('Usage: $0 [options] [input_file...]')
+  .option('version', {
+    describe: 'Show version and exit.',
+    type: 'boolean'
+  })
+  .option('optimize', {
+    describe: 'Enable bytecode optimizer.',
+    type: 'boolean'
+  })
+  .option('bin', {
+    describe: 'Binary of the contracts in hex.',
+    type: 'boolean'
+  })
+  .global([ 'version', 'optimize' ])
+  .version()
+  .showHelpOnFail(false, 'Specify --help for available options')
+  .help()
+  .demand(1, 'Must provide a file');
+
+var argv = yargs.argv;
+var files = argv._;
+
+function abort (msg) {
+  console.log(msg || 'Error occured');
+  process.exit(1);
+}
+
+if (!argv.bin) {
+  abort('Invalid option selected');
+}
+
+var sources = {};
+
+for (var i = 0; i < files.length; i++) {
+  sources[ files[i] ] = fs.readFileSync(files[i]).toString();
+}
+
+var output = solc.compile({ sources: sources }, argv.optimize ? 1 : 0);
+
+for (var contractName in output.contracts) {
+  fs.writeFileSync(contractName + '.bin', output.contracts[contractName].bytecode);
```

Comment:
I'm not so sure it is a good idea to just write to arbitrary files - a clean output directory would probably be better.